autopush/tests/test_integration.py

```diff
@@ -619,6 +619,18 @@ def test_simplepush_data_delivery(self):
         yield self.shut_down(client)
 
     @inlineCallbacks
+    def test_disabled_simplepush(self):
+        state = self.conn.conf.disable_simplepush
+        self.conn.conf.disable_simplepush = True
+        try:
+            yield self.quick_register()
+        except AssertionError as ex:
+            # Check the raised assertion, since the true one is buried.
+            ok_("401 " in ex.message)
+            pass
```

Comment:
kill this line. To be safe I always go with an else: assert False, "Expected AssertionError w/ 401" block on try excepts like this. the last line resetting the flag should be in a finally block too.

the rarely seen try-except-else-finally